yeonjae/programmers-2/숫자의표현.js

```diff
@@ -0,0 +1,18 @@
+function solution(n) {
+  let answer = 1;
+
+  for (let k = 1; k <= n / 2; k++) {
+    let sum = 0;
+    let i = k;
+
+    while (sum < n) sum += i++;
+    if (sum == n) answer++;
+  }
+  return answer;
+}
```

Comment:
```suggestion
function solution(n) {
  let answer = 1;

  for (let i = 1; i <= n / 2; i++) {
    let sum = 0;
    let j = i;

    while (sum < n) {
      sum += j++;
    }
    if (sum === n) answer++;
  }
    
  return answer;
}
```
- 풀이잘봤습니다! 조금 언급하고 싶은게 많은데요..🥲
- ⭐️ 첫번째 for문에서 k를 먼저 정의한 이유가있을까요? k말고 i로 선언하고 그 다음 i를 j로 선언하는 것을 추천드립니다!
- ⭐️ 그리고 "==" 연산자가 아닌 "==="연산자로 이용해주세요...!!
- ⭐️ 또한 while문, for문 등에서는 한 줄 표기법이 일반적인 자바스크립트 컨벤션에 위배되는 행위입니다! 괄호를 만들어주세요
- 굉장히 중요한 부분이라 별표1!!
- 아래와 같은 투포인터 알고리즘을 이용하면 depth1로도 풀이가 가능합니다
```js
function solution(n) {
  let [i, j] = [1, 1];
  let tempValue = 0;
  let result = 0;

  while (i <= n + 1) {
    if (tempValue < n) {
      tempValue += i++;
      continue;
    }
    if (tempValue === n) {
      result++;
    }
    tempValue -= j++;
  }
  return result;
}
```